http/src/lib.rs

```diff
@@ -503,8 +503,9 @@ fn serve<M: jsonrpc::Metadata, S: jsonrpc::Middleware<M>>(
 
 			let mut http = server::conn::Http::new();
 			http.keep_alive(keep_alive);
+			let tcp_stream = SuspendableStream::new(listener.incoming());
```

Comment:
@debris 
1. `incoming` is a stream of incoming sockets
2. Disconnecting has nothing to do with that stream at all
3. Polling the stream may return a temporary `io::Error` (for instance if we can't open the connection because of "too many open files" limit)
4. we use `for_each` combinator which:
4.1. Runs for every `Ok(socket)`
4.2. Stops on the FIRST `Err()`
5. So any temporary `io::Error` will cause the entire server to terminate.
6. The stuff that @seunlanlege is working on causes temporary `io::Error` to sleep a little bit and re-try instead of terminating the server.
7. So the timeout introduced here only affects the server in case of heavy usage conditions, without the timeout the connection would be unsuccesful anyway.

Reply:
The reason we need this is because `Stream::for_each` returns a future that will end, if the underlying stream encounters an error. So far the only error we've seen in the wild is an io error of too many open files. This is a recoverable error and shouldn't terminate the server. which is what currently happens as reported here paritytech/parity-ethereum#9102.

The solution is to wait for a while before accepting any new connections, so we have the chance to complete and drop the existing connections.

Reply:
@tomusdrw @seunlanlege can you add 1, 3, 4, 5 and 6 as the comment to the code? It will greatly help future maintainers of this code :)